pkg/compiler/testdata/all.txt

```diff
@@ -368,6 +368,7 @@ union$conditional3 [
 	u1	int32	(if[value[parent:parent:f0] == 3])
 	u2	int32	(if[value[struct$conditional:f0] == 3])
 	u3	const[0xabcd, int64]
+	u4	int64	(if[value[u3]] == 0xabcd)
```

Comment:
Hmmm, we shouldn't have been able to compile this -- `u3` refers to another member of the same union, so this `if[]` condition does not make any sense.

We do have some checks against unions in the path:
https://github.com/google/syzkaller/blob/9750182a9a67f35e95cb1e077a3b69a4a9b54083/pkg/compiler/check.go#L611-L615

But we apparently assume that the first parent is always a struct (which it isn't in this case).

Reply:
It does fail :-) I missed this was a union :sweat_smile: